features/FEATURE_PSA/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,598 @@
+building PSA API compliance tests.
+
+The
+[mbed-os-tf-m-regression-tests](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests)
+contains build scripts, TF-M regression tests and PSA API compliance tests.
+
+## Building TF-M and running regression tests
+Follow the steps below to build TF-M and regression tests:
+
+* Clone
+  [mbed-os-tf-m-regression-tests](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests)
+  repo
+* Update `mbed-os.lib` to the version of Mbed OS that contains new target
+  support including `targets.json` changes described in this document
+* Run `mbed deploy`
+* Run `python3 build_tfm.py -m <new target> -t <toolchain> -c
+  ConfigRegressionIPC.cmake`
+    * The command builds necessary services in TF-M to run regression tests
+    *NOTE*\
+    Use this config **ONLY TO** run regression tests. The secure binary copied
+    to Mbed OS **MUST** be generated using the config option
+    `ConfigCoreIPC.cmake`.
+* Ensure TF-M binary is copied to location defined by target attribute
+  `tfm_delivery_dir`
+* Run `mbed compile -m <new target> -t <toolchain>` to build Mbed OS and run
+  regression tests
+* Flash the regression tests binary on to the target and ensure all regression
+  tests passes
+
+*TIP*\
+Supported options:
+```console
+usage: build_tfm.py [-h]
+                    [-c {ConfigCoreIPC.cmake,ConfigRegressionIPC.cmake,ConfigPsaApiTestIPC.cmake}]
+                    [-m {ARM_MUSCA_A1,ARM_MUSCA_B1,CY8CKIT_064S2_4343W,CYESKIT_064B0S2_4343W}]
+                    [-t {ARMCLANG,GNUARM}] [-d] [-l] [--commit]
+                    [-s {CRYPTO,INITIAL_ATTESTATION,PROTECTED_STORAGE,INTERNAL_TRUSTED_STORAGE}]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -c {ConfigCoreIPC.cmake,ConfigRegressionIPC.cmake,ConfigPsaApiTestIPC.cmake}, --config {ConfigCoreIPC.cmake,ConfigRegressionIPC.cmake,ConfigPsaApiTestIPC.cmake}
+                        Use the specified TF-M configuration
+  -m {ARM_MUSCA_A1,ARM_MUSCA_B1,CY8CKIT_064S2_4343W,CYESKIT_064B0S2_4343W}, --mcu {ARM_MUSCA_A1,ARM_MUSCA_B1,CY8CKIT_064S2_4343W,CYESKIT_064B0S2_4343W}
+                        Build for the given MCU
+  -t {ARMCLANG,GNUARM}, --toolchain {ARMCLANG,GNUARM}
+                        Build for the given toolchain (default is
+                        tfm_default_toolchain)
+  -d, --debug           Set build profile to debug
+  -l, --list            Print supported TF-M secure targets
+  --commit              Commit secure binaries (TF-M) and
+                        features/FEATURE_PSA/TARGET_TFM/VERSION.txt
+  -s {CRYPTO,INITIAL_ATTESTATION,PROTECTED_STORAGE,INTERNAL_TRUSTED_STORAGE}, --suite {CRYPTO,INITIAL_ATTESTATION,PROTECTED_STORAGE,INTERNAL_TRUSTED_STORAGE}
+                        Suite name for PSA API Tests
+```
+
+## Building TF-M and running PSA compliance tests
+Follow the steps below to build TF-M and compliance tests:
+
+* Switch to `mbed-os-tf-m-regression-tests` directory
```

Comment:
I think there is an instruction missing here - https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/master/main.cpp#L8 , we need to switch the example.

Reply:
That repo defaults to regression tests, which is what we want to run here, isn't it?

Reply:
Fixed